### PR TITLE
Add java-matrix and java-fips-matrix Buildkite pipelines

### PR DESCRIPTION
**Pull request by brianseeders**
- Add a pre-command hook for setting up environment and secrets
- Add periodic Buildkite pipeline (3x daily) with matrix steps for java-fips-matrix and java-matrix, mirroring what Jenkins currently runs
- Don't fail `init.gradle` when running outside of CI (this is mostly useful when running personal VM instances that reproduce the CI environment)
- Switch vault paths depending on the CI environment (Jenkins or Buildkite)
- Fix `init.gradle` settings for build scans (from @mark-vieira)
- Fix permissions on lintian/overrides file for newer versions of lintian that require it

Tested in Buildkite here: https://buildkite.com/elastic/brianseeders-elasticsearch-sandbox/builds?branch=buildkite-migration-java-compat

### CODE REVIEW

File: .buildkite/hooks/pre-command [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# On some distros, this directory ends up not readable by the `elasticsearch` user that gets created during tests
+# This fixes that
+chmod 755 ~
+
+WORKSPACE="$(pwd)"
+export WORKSPACE
```
Comment (mark-vieira) [inline .buildkite/hooks/pre-command:8]: Is there a reason not to just combine these lines into something like `export WORKSPACE="$(pwd)"`?

File: .buildkite/hooks/pre-command [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+JOB_BRANCH="$BUILDKITE_BRANCH"
+export JOB_BRANCH
+
+GRADLEW="./gradlew --parallel --scan --build-cache --no-watch-fs -Dorg.elasticsearch.build.cache.url=https://gradle-enterprise.elastic.co/cache/"
+export GRADLEW
+
+GRADLEW_BAT="./gradlew.bat --parallel --scan --build-cache --no-watch-fs -Dorg.elasticsearch.build.cache.url=https://gradle-enterprise.elastic.co/cache/"
+export GRADLEW_BAT
+
+export $(cat .ci/java-versions.properties | grep '=' | xargs)
```
Comment (mark-vieira) [inline .buildkite/hooks/pre-command:25]: Does `source` not work as it does in our Jenkins setup?

File: .buildkite/pipelines/periodic.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        command: .ci/scripts/run-gradle.sh -Dbwc.checkout.align=true -Dtests.fips.enabled=true $$GRADLE_TASK
+        timeout_in_minutes: 180
+        matrix:
+          setup:
+            ES_RUNTIME_JAVA:
+              - openjdk17
+            GRADLE_TASK:
+              - checkPart1
+              - checkPart2
+              - checkPart3
```
Comment (mark-vieira) [inline .buildkite/pipelines/periodic.yml:14]: Note: However we handle branches, we need to ensure that this task is excluded in 7.17 since it doesn't exist there.

File: .ci/init.gradle [hunk 4]
```diff
@@ -13,14 +13,26 @@ initscript {
 boolean USE_ARTIFACTORY = false
 
 if (System.getenv('VAULT_ADDR') == null) {
+  // When trying to reproduce errors outside of CI, it can be useful to allow this to just return rather than blowing up
```
Comment (mark-vieira) [inline .ci/init.gradle:16]: 👍

File: .ci/scripts/run-gradle.sh [hunk 5]
```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
 # drop page cache and kernel slab objects on linux
 [[ -x /usr/local/sbin/drop-caches ]] && sudo /usr/local/sbin/drop-caches
+
+GRADLE_USER_HOME="$HOME/.gradle"
```
Comment (mark-vieira) [inline .ci/scripts/run-gradle.sh:5]: Why is this needed?

File: catalog-info.yaml [hunk 6]
```diff
@@ -62,3 +62,41 @@ spec:
...
+    kind: Pipeline
+    metadata:
+      description: ":elasticsearch: tests and checks that are run a few times daily"
+      name: elasticsearch / periodic
+    spec:
+      repository: elastic/elasticsearch
+      pipeline_file: .buildkite/pipelines/periodic.yml
+      branch_configuration: main
+      teams:
+        elasticsearch-team: {}
```
Comment (mark-vieira) [inline catalog-info.yaml:90]: Let's add `ml-core` here, as well as the other pipelines.

File: catalog-info.yaml [hunk 7]
```diff
@@ -62,3 +62,41 @@ spec:
...
+      url: https://buildkite.com/elastic/elasticsearch-periodic
+spec:
+  type: buildkite-pipeline
+  system: buildkite
+  owner: group:elasticsearch-team
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      description: ":elasticsearch: tests and checks that are run a few times daily"
```
Comment (mark-vieira) [inline catalog-info.yaml:83]: nit: Can we capitalize `tests` here?

File: .buildkite/hooks/pre-command (hunk 1 above)
Comment (brianseeders) [reply .buildkite/hooks/pre-command:8]: I generally follow `shellcheck` guidelines when writing shell, and it's one of them: https://www.shellcheck.net/wiki/SC2155

File: .buildkite/hooks/pre-command (hunk 2 above)
Comment (brianseeders) [reply .buildkite/hooks/pre-command:25]: I wanted to export them, and it doesn't `source` won't export them. Jenkins uses a separate properties-file inject task to do it.

File: .buildkite/pipelines/periodic.yml (hunk 3 above)
Comment (brianseeders) [reply .buildkite/pipelines/periodic.yml:14]: Yep, this file just won't include the task on the 7.17 branch. The slightly more complicated issue would be if the 7.17 branch wasn't supposed to have the `periodic.yml` pipeline at all.

File: .ci/scripts/run-gradle.sh (hunk 5 above)
Comment (brianseeders) [reply .ci/scripts/run-gradle.sh:5]: It honestly might not be, let me take it out and make sure everything is still good. I think it's left over from when I was debugging an issue that ended up being something else.

Comment (brianseeders) [reply .ci/scripts/run-gradle.sh]: Still good: https://buildkite.com/elastic/brianseeders-elasticsearch-sandbox/builds/134

I'll address it if it pops back up when I'm back to testing other pipelines

**Comment by brianseeders** [comment]
@breskeby could you maybe take a look at this since Mark is out? He initially approved it, and I got his requested changes in, but he wasn't able to re-review before vacation

**Review by breskeby — APPROVED** [review]
lgtm. your shell game is strong!
